pkg/xds/bootstrap/generator.go

```diff
@@ -41,6 +41,7 @@ func NewDefaultBootstrapGenerator(
 	config *bootstrap_config.BootstrapServerConfig,
 	dpServerCertFile string,
 	dpAuthEnabled bool,
+	universal bool,
```

Comment:
not a blocker, but I'd rather do pass here whether the HDS is enabled and then if on that.

(you can explicitly do HDS.enabled=false with Kubernetes deployments like I commented before or you can force HDS.enabled=false on K8S with autoconfigure)